flocker/node/_deploy.py

```diff
@@ -455,14 +447,92 @@ def map_volumes_to_size(volumes):
                         volume.name.dataset_id, volume.size.maximum_size)
             return primary_manifestations
         volumes.addCallback(map_volumes_to_size)
-        d = gatherResults([self.docker_client.list(), volumes])
 
-        def applications_from_units(result):
-            units, available_manifestations = result
-            applications = []
+        def got_volumes(available_manifestations):
             manifestations = []  # Manifestation objects we're constructing
             manifestation_paths = {dataset_id: path for (path, (dataset_id, _))
                                    in available_manifestations.items()}
+
+            manifestations += list(
```

Comment:
It was an empty list before this.  Just `manifestations = ...`, perhaps, and lose `manifestations = []` above?